### ISSUE TITLE
vhost configuration error

### ISSUE THREAD

**Issue by thanosme** [jwilder/nginx-proxy#474]
Hello! I am trying to proxy a seafile server which I also run on a container. I run the proxy container with:
```
docker run -d -p 80:80 -p 443:443 --name nginx-proxy -v nginx_certs:/etc/nginx/certs:ro -v nginx_vhost:/etc/nginx/vhost.d -v nginx_html:/usr/share/nginx/html -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy
```

Then I add this particular configuration at nginx_vhost/seafile.example.com with this content:
```
server {
        listen       80;
        server_name  seafile.example.com;
        rewrite ^ https://$http_host$request_uri? permanent;    # force redirect http to https
    }
    server {
        listen 443;
        ssl on;
        ssl_certificate /etc/ssl/cacert.pem;        # path to your cacert.pem
        ssl_certificate_key /etc/ssl/privkey.pem;    # path to your privkey.pem
        server_name seafile.example.com;
        proxy_set_header X-Forwarded-For $remote_addr;

        add_header Strict-Transport-Security "max-age=31536000; includeSubdomains";
        server_tokens off;

        location / {
            fastcgi_pass    127.0.0.1:8000;
            fastcgi_param   SCRIPT_FILENAME     $document_root$fastcgi_script_name;
            fastcgi_param   PATH_INFO           $fastcgi_script_name;

            fastcgi_param   SERVER_PROTOCOL        $server_protocol;
            fastcgi_param   QUERY_STRING        $query_string;
            fastcgi_param   REQUEST_METHOD      $request_method;
            fastcgi_param   CONTENT_TYPE        $content_type;
            fastcgi_param   CONTENT_LENGTH      $content_length;
            fastcgi_param   SERVER_ADDR         $server_addr;
            fastcgi_param   SERVER_PORT         $server_port;
            fastcgi_param   SERVER_NAME         $server_name;
            fastcgi_param   HTTPS               on;
            fastcgi_param   HTTP_SCHEME         https;

            access_log      /var/log/nginx/seahub.access.log;
            error_log       /var/log/nginx/seahub.error.log;
            fastcgi_read_timeout 36000;
        }
        location /seafhttp {
            rewrite ^/seafhttp(.*)$ $1 break;
            proxy_pass http://127.0.0.1:8082;
            client_max_body_size 0;
            proxy_connect_timeout  36000s;
            proxy_read_timeout  36000s;
            proxy_send_timeout  36000s;
            send_timeout  36000s;
        }
        location /media {
            root /home/user/haiwen/seafile-server-latest/seahub;
        }
    }
```

and restart the container and get this error:
```
forego     | starting nginx.1 on port 5000
forego     | starting dockergen.1 on port 5100
dockergen.1 | 2016/06/07 12:58:14 Generated '/etc/nginx/conf.d/default.conf' from 4 containers
dockergen.1 | 2016/06/07 12:58:14 Running 'nginx -s reload'
dockergen.1 | 2016/06/07 12:58:14 Error running notify command: nginx -s reload, exit status 1
dockergen.1 | 2016/06/07 12:58:14 Watching docker events
dockergen.1 | 2016/06/07 12:58:14 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification 'nginx -s reload'
```

In the wiki it says that per virtual host configuration should be in the vhost directory and should be named after the domain as I did. Am I doing something wrong for this situation? I have successfully run seafile without the vhost configuration, but I need it because there are some location directives that must be declared so that seafile may download/upload files.

**Comment by wader** [comment]
Where do you run seafile, in a container or on the host? if in a container i don't think you need any or very little of that config as nginx-proxy will generate it for you. If you run it on the host you probably want to have that config in a `/etc/nginx.d/conf.d/file.conf` file.

**Comment by thanosme** [comment]
I run it on a container with the options for VIRTUAL_HOST and VIRTUAL_PORT. Just with this it works partially because port 8082 needs to be rewrited for seafile to work correctly. I tried to use only the location directives as seafile.example.com_location but it doesn't seem to work at all.

**Comment by wader** [comment]
Oh so seafile runs two different web servers? not really sure how to do that in a nice way. What kind of web server are seafile using, can it do the rewrite and proxying maybe?

**Comment by thanosme** [comment]
As the vhost file that I posted above the seafile webui runs on port 8000 and the file server on 8082 and the port 8082 is rewritten to seafile.example.com/seafhttp. Isn' t there a way to declare this whole vhost file to the nginx-proxy?

**Comment by wader** [comment]
I think the problem is that all ways of including configuration can only include static configuration and the configuration you need will have to include the dynamic container IP.

Would it be possible to split the server into two containers? but that assume they can run on different host names instead of using `/seafhttp` to route correctly.

Another way i guess would be to do the `/seafhttp` -> 80802 rewrite inside the seafile container somehow... like running an nginx in the container just to do the rewrite routing.

**Comment by thanosme** [comment]
Allright I found a solution so I will post it for future reference. 

I built a custom container for seafile which included seafile as well as a built in nginx proxy.
Using the internal nginx proxy I did all the rewriting that was needed, using the vhost file above for the ports, and managed to expose only one port for seafile. Then using the jwilder/nginx-proxy without extra configuration the container is rechable and I can even use https without any problem.

Example commands:

```
docker run -d -p 80:80 -p 443:443 --name nginx-proxy -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy
```

```
docker run -d --name seafile -e 'VIRTUAL_HOST=seafile.example.org' -e 'VIRTUAL_PORT=8089' -p 127.0.0.1:8089:8089 seafile
```

I will publish the docker image as soon as I can because I think many people will face the same problem.
